bgcval2/timeseries/timeseriesTools.py

```diff
@@ -254,6 +257,69 @@ def getHorizontalSlice(nc, coords, details, layer, data=''):
     assert 0
 
 
+def save_csv(
+        timesD,
+        arrD, 
+        analysisname,
+        colours={},
+        linestyles={},
+        thicknesses={},
+        labels={},
+        name='',
+        units='',
+        region='',
+        layer='',
+        metric='',
+        title='',
+        ts='',
+        csvFolder='',
+        csvformat='.json',
+    ):
+    """
+    Output the data that appears in a plot as a csv.
+
+    """
+    if csvformat.lower() not in ['json', '.json']:
+        print('Format not set up', csvformat )
+        return
+    
+    filename = bvt.folder(csvFolder) + '_'.join([analysisname, name, region, layer, metric, ts ]) + csvformat
+
+    jsondata = {
+            # json can't save numpy.float32, so we convert to list of floats.
+            'timesD': {job:[float(t) for t in times] for job, times in timesD.items()},
+            'arrD': {job:[float(d) for d in data] for job, data in arrD.items()},
+            'analysisname': analysisname,
+            'colours':colours,
+            'linestyles':linestyles,
+            'thicknesses':thicknesses,
+            'labels':labels,
+            'name':name,
+            'units':units,
+            'region':region,
+            'layer':layer,
+            'metric':metric,
+            'title':title,
+            'analysisname': analysisname,
+            'ts':ts,
+            'filename':filename,
+        }
+    
+    if os.path.exists(filename):
+        print('Opening old file:', filename)
+        with open(filename) as json_data:
+            json_old = json.load(json_data)
+       
+        diff = jsondiff(jsondata, json_old)
+        if not len(diff):
+            print('Nothing new to add')
+            return
+
+    print('Data saved in: ', filename)
+    with open(filename, 'w', encoding='utf-8') as f:
+        json.dump(jsondata, f, ensure_ascii=False, indent=4)
```

Comment:
that's fine, do you know the `ujson` package? Could just use that via `ujson.loads` and `ujson.dumps` https://pypi.org/project/ujson/

Reply:
It looks like `ujson` is faster too. Probably worth making the change. But other users have started using the output already, so it will have to produce identical output. Do you know if it's the same?

Reply:
I am 99% sure it's the same (but not 100% sure) - better leave it as it is, your JSONs are not big and not many either

Reply:
okay!